### ISSUE TITLE
Walls should respawn

### ISSUE THREAD

**Issue by deferred-impact** [deferred-impact/battledudes-issue-tracker#140]
Come play battledudes.io! Destroyable everything!

It's a fun gimmick, but it has some issues. Primarily, the fact that after a short amount of time, the entire area is completely wiped clean by explosions and bulletspam. Fighting in an open field is fun in a sniper vs sniper situation, but with any other gun involved it is terrible.

Make walls respawn after some time!

The main concern with this idea is, how do we prevent walls from spawning right in front of you. I see the following solutions:
 - wall respawn timer begins counting down from 30 seconds for any given bit of a wall as soon as it is destroyed
 - if a bullet or a player passes through a destroyed wall, its respawn timer is reset to 10 seconds
 - if a wall's respawn timer is almost done (3 seconds left) it is rendered in some way. Semi-transparent is the obvious choice, but transparency is always troublesome as it can be hard to read. Perhaps the wall could be rendered in a "dotted line" style, if you know what i mean

Whether roofs should regenerate if enough walls regenerate, is an interesting questions. Of course personally, i want the roofs to remain deleted forever, or at least until the end of the game.

**Comment by Gregory-Y** [comment]
Good idea, I would also add that nearly destroyed walls should have a different model that has cracks or something else that indicates damage, so players know when their cover is about to be destroyed.
